File: CS/Dashboard_BindingToList/Form1.cs
using DevExpress.DashboardCommon;
using DevExpress.DashboardWin;
using DevExpress.DataAccess;
using DevExpress.XtraEditors;
using System;

namespace Dashboard_BindingToList {
    public partial class Form1 : XtraForm {
        public Form1() {
            InitializeComponent();
            dashboardViewer1.DataSourceOptions.ObjectDataSourceLoadingBehavior = DocumentLoadingBehavior.LoadAsIs;
            dashboardViewer1.CustomizeDashboardTitle += DashboardViewer1_CustomizeDashboardTitle;
        }

        private void Form1_Load(object sender, System.EventArgs e) {
            Dashboard dashboard = new Dashboard();

            DashboardObjectDataSource dataSource = new DashboardObjectDataSource("Data Source 1");
            dashboardViewer1.DataLoading += (s, ev) => {
                if (ev.DataSourceName == "Data Source 1")
                    ev.Data = Data.CreateData();
            };
            dashboard.DataSources.Add(dataSource);

            PieDashboardItem pies = new PieDashboardItem();
            pies.DataSource = dashboard.DataSources[0];
            Dimension salesPersonArgument = new Dimension("SalesPerson");
            Measure quantity = new Measure("Quantity");
            pies.Arguments.Add(salesPersonArgument);
            salesPersonArgument.TopNOptions.Enabled = true; salesPersonArgument.TopNOptions.Count = 3;
            salesPersonArgument.TopNOptions.Measure = quantity;
            pies.Values.Add(quantity);

            GridDashboardItem grid = new GridDashboardItem();
            grid.DataSource = dashboard.DataSources[0];
            grid.Columns.Add(new GridDimensionColumn(new Dimension("SalesPerson")));
            grid.Columns.Add(new GridMeasureColumn(new Measure("Quantity")));

            dashboard.Items.AddRange(pies, grid);
            dashboardViewer1.Dashboard = dashboard;

        }

        private void DashboardViewer1_CustomizeDashboardTitle(object sender, CustomizeDashboardTitleEventArgs e)
        {
            DashboardToolbarItem titleButton = new DashboardToolbarItem("Load Data",
                new Action<DashboardToolbarItemClickEventArgs>((args) =>
                {
                    dashboardViewer1.ReloadData();
                }));
            titleButton.Caption = "Reload Data";
            e.Items.Add(titleButton);
        }
    }
}

File: VB/Dashboard_BindingToList/Form1.vb
Imports DevExpress.DashboardCommon
Imports DevExpress.XtraEditors

Namespace Dashboard_BindingToList

    Public Partial Class Form1
        Inherits XtraForm

        Public Sub New()
            InitializeComponent()
        End Sub

        Private Sub Form1_Load(ByVal sender As Object, ByVal e As System.EventArgs)
            Dim dashboard As Dashboard = New Dashboard()
            dashboard.AddDataSource("Data Source 1", Data.CreateData())
            ' Creates a Pie dashboard item that displays share of sold units quantity per sales person.
            Dim pies As PieDashboardItem = New PieDashboardItem()
            pies.DataSource = dashboard.DataSources(0)
            Dim salesPersonArgument As Dimension = New Dimension("SalesPerson")
            Dim quantity As Measure = New Measure("Quantity")
            pies.Arguments.Add(salesPersonArgument)
            salesPersonArgument.TopNOptions.Enabled = True
            salesPersonArgument.TopNOptions.Count = 3
            salesPersonArgument.TopNOptions.Measure = quantity
            pies.Values.Add(quantity)
            ' Creates a Grid dashboard item that displays sales persons and corresponding quantities.
            Dim grid As GridDashboardItem = New GridDashboardItem()
            grid.DataSource = dashboard.DataSources(0)
            grid.Columns.Add(New GridDimensionColumn(New Dimension("SalesPerson")))
            grid.Columns.Add(New GridMeasureColumn(New Measure("Quantity")))
            dashboard.Items.AddRange(pies, grid)
            dashboardViewer1.Dashboard = dashboard
        End Sub

        ' Handles the DashboardViewer.DataLoading event to provide the dashboard with new data.
        Private Sub dashboardViewer1_DataLoading(ByVal sender As Object, ByVal e As DevExpress.DataAccess.DataLoadingEventArgs)
            If Equals(e.DataSourceName, "Data Source 1") Then e.Data = Data.CreateData()
        End Sub

        Private Sub button1_Click(ByVal sender As Object, ByVal e As System.EventArgs)
            ' Reloads data in data sources.
            dashboardViewer1.ReloadData()
        End Sub
    End Class
End Namespace
